Avoid duplicate resource for install directory

Etherdaemon opened the pull request:
Hi @joshbeard,

Firstly thank you for this module. So far it's been working fabulously.

May I propose the following PR to allow users to create their own install directories in other manifests?

My own use case is that I want my install directory to be /opt/bamboo but I need it to be a link to /mnt/ebs_data which means I need to manage that particular directory myself.

Cheers.

Etherdaemon committed:
Avoid duplicate resource for install directory

joshbeard commented:
Sure, that'd be useful.  However, I'd rather avoid `defined()` due to evaluation order (https://docs.puppetlabs.com/puppet/latest/reference/function.html#defined).  How about a parameter to toggle whether the Bamboo module should manage the installation directory itself or not?

Etherdaemon committed:
Update to utilise input parameters - also allow user to specify custom app directory is required

Etherdaemon commented:
Hi @joshbeard - I understand - I've updated the PR to be checking for a true/false parameter instead.

I've also added the option for a user override for the application directory - hope that's all ok?

Cheers

joshbeard committed:
Tune PR #4

Expand upon @Etherdaemon's PR to allow custom management of the
installdir and appdir.

- This updates the README, CHANGELOG, and metadata.
- Change 'app_dir' to 'appdir' to be consistent with the other param
names.
- Remove the resource relationship that requires the app_dir file
resource from the staging::file - this way 'app_dir' could be a mount or
whatever.

joshbeard commented:
Thanks, @Etherdaemon.  I made a couple of minor adjustments and sent you a PR on this to look over.  https://github.com/Etherdaemon/puppet-bamboo/pull/1  If you agree with those, please merge and I'll merge this in.

Etherdaemon commented:
Thanks @joshbeard - merged and pushed update back - thank you very much for taking a look at this
